packaging/Dockerfile.mockbuild

```diff
@@ -6,7 +6,7 @@
 
 # Pull base image
 FROM fedora:latest
-LABEL maintainer="daos@daos.groups.io>"
+MAINTAINER daos-stack <daos@daos.groups.io>
```

Comment:
We are tracking daos-stack/packaging#60 for packaging at the moment.  That will land to master once it's seen a bit more testing.  But updating packaging here in this PR from packaging's master has rolled back a bunch of needed changes.

Reply:
Not sure I follow what is requested here. I've updated packaging/ due to previous PR complaining that packaging was out of date.

Reply:
Simply checkout the bmurrell/leap15-build-should-be-15.3 branch in the packaging repo and update the packaging here from that.  All of the packaging/ changes in this PR are actually reverting the newest packaging changes in this repo which came from that branch.

Reply:
done. based on discussion in chat, updated packaging from the latest master as changes have just landed there